### PR TITLE
Declare license and copyright information using REUSE

### PR DESCRIPTION
**Pull request by msohn**
We now use https://reuse.software/ to declare license and copyright
information in a machine readable format.

The copyright statement for SAP open source projects was updated
to give credit to project contributors not employed by SAP.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```noteworthy developer
License and copyright information is now specified in REUSE format.
```

### CODE REVIEW

File: .reuse/dep5
Comment (rfranzke) [inline .reuse/dep5:1]: How do you generate this file, btw?

**Review by rfranzke — APPROVED** [review]
/lgtm

File: .reuse/dep5
Comment (msohn) [reply .reuse/dep5:1]: The first part about our own component is written manually since it depends on project sources.

For the 3rd party source dependencies I currently retrieve the basic license information for vendored
golang dependencies using askalono [1] and jq [2] like this:

askalono --format json crawl vendor/ \
  | sort \
  | jq -r '{ Files: .path,
             Copyright: "",
             License: .result.license.name }
          | select(.Files | test("/license[.a-zA-Z]*$"; "i"))
          | "Files: \(.Files | sub("(?<p>.*)/license[.a-zA-Z]*$"; "\(.p)/*"; "i"))\nCopyright: \(.Copyright)\nLicense: \(.License)\n"' >> .reuse/dep5

then I run scancode-toolkit [3] to get copyright information and copy that manually to .reuse/dep5 using the scancode workbench UI [4] since it's hard to predict which copyright from which file to use. If it's in LICENSE or NOTICE then it's clear but only a fraction of components does this.

I plan to get both license and copyright information from scancode, though this needs a bit more jq work
since the output format of this tool is more complex. An ideal solution would implement export to debian copyright format as a scancode-toolkit plugin [5].

I am working on documentation how this is done so that we can distribute this work across the team.

[1] https://github.com/jpeddicord/askalono
[2] https://github.com/stedolan/jq
[3] https://github.com/nexB/scancode-toolkit
[4] https://github.com/nexB/scancode-workbench
[5] https://scancode-toolkit.readthedocs.io/en/latest/plugins/plugin_arch.html

Comment (rfranzke) [reply .reuse/dep5:1]: Well, even with documentation, you cannot expect that anybody will maintain this. It wasn't done in the past and it will most likely not be done in the future, there are simply too many repositories, and it's too cumbersome work. The only sustainable solution is to have a script that does this and updates this file on every `make revendor` execution. I don't want to hold this PR, so we can get it in as is, but I want to set the expectations that it will become again outdated very soon if we don't have automation for it.
Do you want to add such script now before the PR is merged or later?

Comment (msohn) [reply .reuse/dep5:1]: I now did this on a couple of the gardener repositories with this semi-automated approach to get a feeling what we need in order to do this properly.

I'll also give it a time-boxed try if I can come up with a script automatically getting license and copyright information from scancode. I also filed a feature request to scancode [1] requesting an implementation there.
If they seem responsive I can try to implement it as a scancode plugin.

I will discuss the experience I gathered so far with OSPO and also ask them for help with implementing proper tool support.

[1] https://github.com/nexB/scancode-toolkit/issues/2235

Comment (rfranzke) [reply .reuse/dep5:1]: OK, so shall we wait until, let's say end of the week, before merging this, or shall we go ahead?

Comment (msohn) [reply .reuse/dep5:1]: I'd say go ahead and merge this, I will discuss the issue about effort to keep information for dependencies up-to-date with OSPO and when we have guidance I will follow-up.

My feature request for scancode got positive feedback from the maintainer, maybe that's the path to a proper solution.
https://github.com/nexB/scancode-toolkit/issues/2235

Comment (rfranzke) [reply .reuse/dep5:1]: Alright, thank you!

**Pull request merged**
